README.md

```diff
@@ -71,7 +71,42 @@ for (const [i, v] of tree.entries()) {
 
 In practice this might be done in a frontend application prior to submitting the proof on-chain, with the address looked up being that of the connected wallet.
 
-See [`MerkleProof`] for documentation on how to validate the proof in Solidity.
+### Validating a Proof in Solidity
+
+Once you've gotten the proof, it can be validated in Solidity using [`MerkleProof`] with the following example:
+
+```solidity
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/utils/cryptography/MerkleProof.sol";
+
+contract Verifier {
+    bytes32 private root;
+
+    constructor(bytes32 _root) {
+        // (1)
+        root = _root;
+    }
+
+    function verify(
+        bytes32[] memory proof,
+        address addr,
+        uint256 amount
+    ) public view returns (bool) {
+        // (2)
+        bytes32 leaf = keccak256(bytes.concat(keccak256(abi.encode(addr, amount))));
+        // (3)
+        return MerkleProof.verify(proof, root, leaf);
```

Comment:
Normally the proof will be verified as a prerequisite to doing some action, so let's add a fourth item where we say there would be some other thing happening (not sure how best to express that).

There's something we should also mention, though we don't have a great recommendation yet, but usually you want to mark a leaf as "used" so it can't be reused. We can just put a note in parentheses saying to consider that there may need be a mechanism to prevent reuse of a leaf/proof.

```suggestion
    ) public {
        // (2)
        bytes32 leaf = keccak256(bytes.concat(keccak256(abi.encode(addr, amount))));
        // (3)
        require(MerkleProof.verify(proof, root, leaf), "Invalid proof");
        // (4)
        // ...
```

Reply:
> let's add a fourth item where we say there would be some other thing happening
Done, however, I decided to not use a `require` and instead express the verification as a `bool verified` value. I think it fits better in expressing that the value might be used for further operations. Let me know if that works.

> We can just put a note in parentheses saying to consider that there may need be a mechanism to prevent reuse of a leaf/proof.
Added!